internal/services/network/client/client.go

```diff
@@ -73,7 +73,7 @@ func NewClient(o *common.ClientOptions) (*Client, error) {
 	configurationPolicyGroupClient := network.NewConfigurationPolicyGroupsClientWithBaseURI(o.ResourceManagerEndpoint, o.SubscriptionId)
 	o.ConfigureClient(&configurationPolicyGroupClient.Client, o.ResourceManagerAuthorizer)
 
-	customIpPrefixesClient := network.NewCustomIPPrefixesClient(o.SubscriptionId)
+	customIpPrefixesClient := network.NewCustomIPPrefixesClientWithBaseURI(o.ResourceManagerEndpoint, o.SubscriptionId)
```

Comment:
@magodo whilst this won't be an issue for much longer given `go-azure-sdk` requires passing the endpoint always - it might be worth adding a bash check ([example](https://github.com/hashicorp/terraform-provider-azurerm/blob/40844b968714a6fb4421c1a4c42b7e1b385b9709/scripts/run-gradually-deprecated.sh#L13-L22)) for the line `Client(o.SubscriptionId)` to catch this in the interim, WDYT?